Well look at that, your nit made me learn something.

---

Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/ui/src/views/cluster/containers/clusterOverview/index.tsx, line 17 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20669#-L0CjFuKrMghU-iDaqQ5:-L0FQZ7aWQwyINLGmjf9:bc1hxza) ([raw file](https://github.com/cockroachdb/cockroach/blob/148869944c32f9f63e5e7e773e4d6dcf33afd74c/pkg/ui/src/views/cluster/containers/clusterOverview/index.tsx#L17)):*
<details><summary><i>Previously, vilterp (Pete Vilter) wrote…</i></summary><blockquote>

does `import spinner from "assets/spinner.gif"` work?
</blockquote></details>

TBH I just cargo-culted that 😳.

Tried it out.  Builds fine (Webpack gets it), but tslint (and maybe tsc) doesn't get it:
`Cannot find module 'assets/spinner.gif'.`

...

UPDATE: after reading through Microsoft/TypeScript#2709 I've found a solution!

I'm not exactly clear on how the `import "*.styl"` was already working, but perhaps because it was just imported for the side-effect, TS doesn't bother checking for it?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20669#-:-L0FUtjBShpE1cx4dHEd:b-ppmt6b)*
<!-- Sent from Reviewable.io -->

--- a/pkg/ui/src/views/cluster/containers/clusterOverview/cluster.styl
+++ b/pkg/ui/src/views/cluster/containers/clusterOverview/cluster.styl
@@ -4,13 +4,16 @@
 .cluster-overview
   /* TODO(couchand): Update the layout styles to get rid of this hack. */
   margin-top -1 * $page-padding-top
-  border-bottom 1px solid $table-border-color
 
   .cluster-summary
     background-color white
     padding-top $page-padding-top
     padding-bottom $page-padding-top
     padding-left 20px
+    margin-left 24px
+    margin-right 24px
+    border 1px solid $table-border-color
+
     color gray
     display grid
     align-items end

--- a/pkg/ui/src/views/cluster/containers/clusterOverview/cluster.styl
+++ b/pkg/ui/src/views/cluster/containers/clusterOverview/cluster.styl
@@ -6,10 +6,6 @@
   margin-top -1 * $page-padding-top
   border-bottom 1px solid $table-border-color
 
-  .cluster-ticker
-    padding-top 0
-    padding-bottom 0
-
   .cluster-summary
     background-color white
     padding-top $page-padding-top

--- a/pkg/ui/src/views/cluster/containers/clusterOverview/index.tsx
+++ b/pkg/ui/src/views/cluster/containers/clusterOverview/index.tsx
@@ -23,7 +23,7 @@ class ClusterTicker extends React.Component<{}, {}> {
   render() {
     return (
       <section className="section cluster-ticker">
-        <h2>Cluster Overview</h2>
+        <h1>Cluster Overview</h1>
       </section>
     );
   }

--- a/pkg/ui/src/views/cluster/containers/clusterOverview/index.tsx
+++ b/pkg/ui/src/views/cluster/containers/clusterOverview/index.tsx
@@ -13,6 +13,9 @@ import capacityChart from "./capacity";
 
 import "./cluster.styl";
 
+// tslint:disable-next-line:no-var-requires
+const spinner = require<string>("assets/spinner.gif");
+
 // tslint:disable-next-line:variable-name
 const CapacityChart = createChartComponent(capacityChart);
 
@@ -151,15 +154,23 @@ interface ClusterSummaryProps {
   capacityUsage: CapacityUsageProps;
   nodeLiveness: NodeLivenessProps;
   replicationStatus: ReplicationStatusProps;
+  loading: boolean;
 }
 
 class ClusterSummary extends React.Component<ClusterSummaryProps, {}> {
   render() {
-    const children = [
-      ...renderCapacityUsage(this.props.capacityUsage),
-      ...renderNodeLiveness(this.props.nodeLiveness),
-      ...renderReplicationStatus(this.props.replicationStatus),
-    ];
+    const children = [];
+
+    if (this.props.loading) {
+      children.push(<img className="visualization__spinner" src={spinner} />);
+    } else {
+      children.push(
+        ...renderCapacityUsage(this.props.capacityUsage),
+        ...renderNodeLiveness(this.props.nodeLiveness),
+        ...renderReplicationStatus(this.props.replicationStatus)
+      );
+    }
+
     return <section className="cluster-summary" children={children} />;
   }
 }
@@ -169,6 +180,7 @@ function mapStateToClusterSummaryProps (state: AdminUIState) {
     capacityUsage: mapStateToCapacityUsageProps(state),
     nodeLiveness: mapStateToNodeLivenessProps(state),
     replicationStatus: mapStateToReplicationStatusProps(state),
+    loading: !state.cachedData.nodes.data,
   };
 }
 

